compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -183,6 +182,18 @@ object Types {
       loop(this)
     }
 
+    /** True iff `symd` is a denotation of a class type parameter and the reference
+     *  `<this> . <symd>` is an actual argument reference, i.e. `this` is different
+     *  from the ThisType of `symd`'s owner.
+     */
+    def isArgPrefix(symd: SymDenotation)(implicit ctx: Context) =
```

Comment:
Suggestion: call the method `isArgPrefixOf`.

Reply:
Good point. In fact, I still want to try something else: Make internal class parameter references take a `NoPrefix`. Then any class parameter reference with some other prefix would be considered an external (type-arg) reference.